### PR TITLE
Fix horizontal trim for block doc comments

### LINKED ISSUE #93309: Generated documentation includes bullet points that are not part of the docs
Hi, I've just published a new version of my crate and the generated documentation is including a lot of bullet points that have never been included before.

https://docs.rs/easy-ml/1.8.1/easy_ml/index.html
https://docs.rs/easy-ml/1.8.0/easy_ml/index.html

When I generate the docs locally on the 1.8.1 source code using the latest stable version of rust I see the same formatting for the 1.8.0 that docs.rs generated, so I think this is a bug that's slipped into the nightly version of docs.rs?

Stable local docs:
Screenshot 2022-01-25 at 22-08-57 easy_ml - Rust
docs.rs:
Screenshot 2022-01-25 at 22-08-43 easy_ml - Rust

My guess is that it's because I'm using block comments, the unwanted bullet points seem to match up with empty lines in the documentation comment which consist of just a `*`.

```rust
/*!
 * If this is your first time using Easy ML you should check out some of the examples
 * to get an overview of how to use matrices then study the
 * [Matrix](matrices::Matrix) type for what you need.
 *
 * # Examples
 * - [Linear Regression](linear_regression)
 * - [k-means Clustering](k_means)
 * - [Logistic Regression](logistic_regression)
 * - [Naïve Bayes](naive_bayes)
 * - [Neural Network XOR Problem](neural_networks)
 *
 * # API Modules
 * - [Matrices](matrices)
 * - [Linear Algebra](linear_algebra)
 * - [Distributions](distributions)
 * - [(Automatic) Differentiation](differentiation)
 * - [Numerical type definitions](numeric)
 *
 * # Miscellaneous
 * - [Web Assembly](web_assembly)
 * - [SARSA and Q-learning using a Matrix for a grid world](sarsa)
 * - [Using custom numeric types](using_custom_types)
 */
```

Thanks

### LINKED ISSUE #93662: Doctest fail to compile, but there is nothing to test
When running the doc tests, I get the following error message:

```
running 2 tests
test src/core.rs - core::Wormhole::send_json (line 269) ... FAILED
test src/core.rs - core::Wormhole::connect_without_code (line 119) ... FAILED

failures:

---- src/core.rs - core::Wormhole::send_json (line 269) stdout ----
error: expected expression, found `}`
 --> src/core.rs:271:1
  |
4 | } _doctest_main_src_core_rs_269_0() }
  | ^ expected expression

error: aborting due to previous error

Couldn't compile the test.
---- src/core.rs - core::Wormhole::connect_without_code (line 119) stdout ----
error: expected expression, found `}`
 --> src/core.rs:121:1
  |
4 | } _doctest_main_src_core_rs_119_0() }
  | ^ expected expression

error: aborting due to previous error

Couldn't compile the test.

failures:
    src/core.rs - core::Wormhole::connect_without_code (line 119)
    src/core.rs - core::Wormhole::send_json (line 269)

test result: FAILED. 0 passed; 2 failed; 0 ignored; 0 measured; 0 filtered out; finished in 0.03s

error: test failed, to rerun pass '--doc'
```

[Source code](https://github.com/magic-wormhole/magic-wormhole.rs/blob/906b3d8fa46dbccfe51cd5510679b9c1b699d24f/src/core.rs) (not minimized yet). The relevant doc comment looks like this:

```rust
    /**
     * Generate a code and connect to the rendezvous server.
     *
     * # Returns <-- this is the line shown in the error message
     *
     * A tuple with a [`WormholeWelcome`] and a [`std::future::Future`] that will
     * do the rest of the client-client handshake and yield the [`Wormhole`] object
     * on success.
     */
```

### Meta

`rustc --version --verbose`:
```
rustc 1.60.0-nightly (4e8fb743c 2022-02-03)
binary: rustc
commit-hash: 4e8fb743ccbec27344b2dd42de7057f41d4ebfdd
commit-date: 2022-02-03
host: x86_64-unknown-linux-gnu
release: 1.60.0-nightly
LLVM version: 13.0.0
```

Some kind soul bisected this onto https://github.com/rust-lang/rust/compare/5e57faa78...777bb86bc (between 2022-01-20 and 2022-01-21)

### PR DESCRIPTION
**Pull request by GuillaumeGomez**
Fixes #93662.
Fixes #93309.

r? @notriddle

### CODE REVIEW

**Comment by GuillaumeGomez** [comment]
From the [commonmark dingus](https://spec.commonmark.org/dingus/?text=%20%20%20%20%20a), the updated rustdoc test should have been a code block in the first place (so still not a list).

File: src/test/rustdoc/strip-block-doc-comments-stars.docblock.html [hunk 1]
```diff
@@ -1,2 +1,6 @@
-<div class="docblock"><p>a</p>
+<div class="docblock"><ul>
+<li>
```
Comment (notriddle) [inline src/test/rustdoc/strip-block-doc-comments-stars.docblock.html:2]: > The goal of this test is to ensure that it won't be generated as a list

It’s generating a list, and it really looks like it shouldn’t.

Comment (GuillaumeGomez) [reply src/test/rustdoc/strip-block-doc-comments-stars.docblock.html:2]: Oh indeed. I only saw the code block. Good catch!

**Comment by GuillaumeGomez** [comment]
Should be better now. :)

**Comment by GuillaumeGomez** [comment]
With fmt it's better. :)

**Review by notriddle — APPROVED** [review]

**Comment by petrochenkov** [comment]
@GuillaumeGomez 
The test `src/test/rustdoc-ui/block-doc-comment.stdout` is causing spurious failures.
It has this line
```
test result: ok. 0 passed; 0 failed; 0 ignored; 0 measured; 0 filtered out; finished in 0.00s
```
but the time is not always 0.00s, it sometimes takes longer.
It needs to be normalized to `$TIME`, like in other similar tests.

**Comment by GuillaumeGomez** [comment]
Good catch! Sending a fix right away!

**Pull request merged**